designs/2023-rule-performance-dashboard/README.md

```diff
@@ -0,0 +1,261 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR: <https://github.com/eslint/rfcs/pull/108>
+- Authors: Mara Nikola Kiefer (@mnkiefer)
+
+# Performance Statistics
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes a new ESLint flag `stats`, which enables the user to obtain a series of runtime *statistics* on top of their final lint results (refer to the table below).
+
+These statistics include more granular timing information, such as the parse-, fix-, and lint-times as well as the number of directives, fix passes, and violations encountered.
+
+| Stats |  | Description |
+|--------|-------|---|
+| **times*** | | The times/time objects spent on (parsing, fixing, linting) a file. |
+|        | fix* | An array of time objects for each of the fix passes. |
+|        | parse | The time that is spent when parsing a file. |
+|        | rules* | An array of time objects for each file/rule/selector combination. |
+| **fixPasses** | | The number of times ESLint has applied at least one fix after linting. |
+| **directives** | | The number of disable directives which have been applied to silence rule violations. |
+| **violations** | | The number of times a rule has been violated. |
+
+> All object properties marked with a `*` also contain a computed `total` property.
+
+A *proof-of-concept* can be found at:
+- [**ESLint**](https://github.com/mnkiefer/eslint/pull/1): **Fork of ESLint** on which the [detailed design](#detailed-design) (as described below) has been implemented.
+- [**Sample project**](https://github.com/mnkiefer/eslint-samples): Project on which the above implementation can be demonstrated.
+
+Below is an brief example on what the `stats` properties would look like (based on the sample project file _invalid/no-regex-spaces_).<br>
+Each property will be addressed individually in the [Detailed Design](#detailed-design) section 2 ("Adding the `stats` option to the Linter").
+
+```json
+"stats": {
+    "directives": 1,
+    "fixPasses": 1,
+    "times": {
+        "parse": 0.310625,
+        "total": 0.35879099999999997,
+        "rules": {
+            "total": 0.044916000000000005,
+            "valid-typeof": {
+                "total": 0.044916000000000005,
+                "UnaryExpression": 0.04079100000000001,
+                "Program": 0.004125
+            }
+        },
+        "fix": {
+            "1": 0.00325,
```

Comment:
How about having an array for passes? An array is a more convenient format to work with (avoids having to parse and check the object keys to get the list), and storing this in `passes` gives the array data a helpful label and leaves room for other arrays to be added later.

```suggestion
            "passes": [0.00325],
```

Reply:
I fully agree. 👍 I've just updated the implementation and example.